libs/testing/fixtures/src/lib/application.ts

```diff
@@ -82,3 +97,145 @@ export const createApplication = (
   status: ApplicationStatus.IN_PROGRESS,
   ...overrides,
 })
+
+type Schema = z.ZodObject<any>
+
+interface Template<
+  TContext extends ApplicationContext,
+  TStateSchema extends ApplicationStateSchema<TEvents>,
+  TEvents extends EventObject
+> {
+  readyForProduction?: boolean
+  featureFlag?: Features
+  type: ApplicationTypes
+  name: string
+  institution: string
+  translationNamespaces?: string[]
+  allowedDelegations?: AuthDelegationType[]
+  dataSchema: Schema
+  stateMachineConfig: MachineConfig<TContext, TStateSchema, TEvents> & {
+    states: StatesConfig<TContext, TStateSchema, TEvents> // TODO Extend StatesConfig to completely enforce meta being required attribute
+  }
+  stateMachineOptions?: Partial<MachineOptions<TContext, TEvents>>
+  mapUserToRole(
+    nationalId: string,
+    application: Application,
+  ): ApplicationRole | undefined
+  answerValidators?: Record<string, AnswerValidator>
+}
+
```

Comment:
I think we should create a new type libs for application (libs/application/types) and move the types so we are not duplicating the Application and Template interface here.